compiler/rustc_metadata/src/rmeta/decoder/cstore_impl.rs

```diff
@@ -203,7 +203,11 @@ impl IntoArgs for (CrateNum, SimplifiedType) {
 }
 
 provide! { tcx, def_id, other, cdata,
-    explicit_item_bounds => { table_defaulted_array }
+    explicit_item_bounds => {
```

Comment:
Similar to https://github.com/rust-lang/rust/pull/110498#discussion_r1170427717, there is some ugly stuff here around skipping the binder for metadata encoding and re-wrapping it for decoding.

I decided to use `EarlyBinder<&'tcx [Predicate<'tcx>, Span]>` for the query output to match what's done in `bound_explicit_item_bounds`, and because there are some nice APIs on `EarlyBinder` that seem designed for this use case (like [`subst_iter_copied`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/subst/struct.EarlyBinder.html#method.subst_iter_copied)).

Using `&'tcx [EarlyBinder<Predicate<'tcx>>, Span]` instead would simplify these conversions around the metadata, but might make some other uses of the query a bit more verbose. Happy to change it if it makes more sense this way though